lib/git-clone.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const child = require('child_process');
+const path = require('path');
+
+const mkdirp = require('mkdirp');
+
+function getExecGit(path) {
+  return function execGit(args) {
+    child.execFileSync('git', args, {
+      cwd: path,
+      stdio: 'ignore'
+    });
+  };
+}
+
+function gitClone(context, next) {
+  const gitUrl = context.module.raw;
+  const gitRef = context.module.ref;
+
+  const modulePath = path.join(context.path, context.module.name);
+  mkdirp.sync(modulePath);
+  const execGit = getExecGit(modulePath);
+
+  try {
+    execGit(['init']);
+    execGit(['remote', 'add', 'origin', gitUrl]);
+    execGit(['fetch', '--depth=1', 'origin', gitRef]);
+    execGit(['checkout', 'FETCH_HEAD']);
```

Comment:
Alternatively: `checkout --recurse-submodules FETCH_HEAD` (instead of `submodule init` and `update`)

Reply:
oh nice, I didn't know about this option 👍

Reply:
Actually, it doesn't work. The git manual says "Using --recurse-submodules will update the content of all initialized submodules according to the commit recorded in the superproject", so it seems we have to `init`. Then it's easier to update.

Reply:
My bad, I assumed it meant the same as in `git clone --recurse-submodules`.